src/cli-types.ts

```diff
@@ -0,0 +1,20 @@
+import {AWSRegion} from './aws-regions';
+import * as yargs from 'yargs';
+
+export {Argv, Options} from 'yargs';
+
+export interface GlobalArguments {
+  region?: AWSRegion;
+  profile?: string;
+  assumeRoleArn?: string;
+  debug?: boolean;
+  logFullError?: boolean;
+  environment: string;
+  clientRequestToken?: string;
+}
+
+export type GenericCLIArguments = GlobalArguments & yargs.Arguments;
+
+export type ExitCode = number;
```

Comment:
Unrelated to the refactor, but could this be limited to the values from `statusCodes`?